ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/CommandTopicTest.java

```diff
@@ -0,0 +1,138 @@
+  public void shouldGetNewCommandsCorrectly() {
+    // Given:
+    final CommandId commandId1 = Mockito.mock(CommandId.class);
+    final Command command1 = Mockito.mock(Command.class);
+    final QueuedCommandStatus queuedCommandStatus1 = Mockito.mock(QueuedCommandStatus.class);
+    final Map<CommandId, QueuedCommandStatus> commandStatusMap = new HashMap();
+    commandStatusMap.put(commandId1, queuedCommandStatus1);
+    final TopicPartition topicPartition = new TopicPartition("foo", 1);
+    final ConsumerRecord consumerRecord = Mockito.mock(ConsumerRecord.class);
+    Mockito.when(consumerRecord.key()).thenReturn(commandId1);
+    Mockito.when(consumerRecord.value()).thenReturn(command1);
+    Mockito.when(commandConsumer.poll(Mockito.any(Duration.class))).thenReturn(new ConsumerRecords(Collections.singletonMap(topicPartition, Collections.singletonList(consumerRecord))));
+
+    // When:
+    final List<QueuedCommand> newCommands = commandTopic.getNewCommands(commandStatusMap);
+
+    // Then:
+    MatcherAssert.assertThat(newCommands.size(), CoreMatchers.equalTo(1));
+    MatcherAssert.assertThat(newCommands.get(0).getCommandId(), CoreMatchers.is(commandId1));
+    MatcherAssert.assertThat(newCommands.get(0).getStatus().get(), CoreMatchers.is(queuedCommandStatus1));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldGetRestoreCommandsCorrectly() {
+    // Given:
+    final CommandId createId = new CommandId(CommandId.Type.TABLE, "one", CommandId.Action.CREATE);
+    final CommandId dropId = new CommandId(CommandId.Type.TABLE, "one", CommandId.Action.DROP);
+    final KsqlConfig ksqlConfig = new KsqlConfig(Collections.emptyMap());
+    final Command originalCommand = new Command(
+        "some statement", Collections.emptyMap(), ksqlConfig.getAllConfigPropsWithSecretsObfuscated());
+    final Command dropCommand = new Command(
+        "drop", Collections.emptyMap(), ksqlConfig.getAllConfigPropsWithSecretsObfuscated());
+    final Command latestCommand = new Command(
+        "a new statement", Collections.emptyMap(), ksqlConfig.getAllConfigPropsWithSecretsObfuscated());
+
+    final ConsumerRecords<CommandId, Command> records = new ConsumerRecords<>(
+        Collections.singletonMap(new TopicPartition("topic", 0), Arrays.asList(
+            new ConsumerRecord<>("topic", 0, 0, createId, originalCommand),
+            new ConsumerRecord<>("topic", 0, 0, dropId, dropCommand),
+            new ConsumerRecord<>("topic", 0, 0, createId, latestCommand))
+    ));
+    Mockito.when(commandConsumer.poll(Mockito.any(Duration.class)))
+        .thenReturn(records)
+        .thenReturn(new ConsumerRecords(Collections.emptyMap()));
+
+    // When:
+    final RestoreCommands restoreCommands = commandTopic.getRestoreCommands(Duration.ofMillis(1));
+
+
+    // Then:
+    assertThat(restoreCommands.getToRestore().size(), equalTo(3));
+    assertThat(restoreCommands.getToRestore().keySet(), equalTo(ImmutableSet.of(
```

Comment:
Try to avoid making things public just for testing. You can make use of the `RestoreCommands.forEach` method to capture the output.   However, Rohan's pending PR removes `RestoreCommands` all together... so talk to him first before spending ages on this.